src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicator.java

```diff
@@ -0,0 +1,1885 @@
+			rGainControl(RGainControl.DOWN);
+			return;
+		}
+
+		// BGain
+		if (splitProperty[1].equals(Command.BGAIN.getName() + BGainControl.UP.getName())) {
+			bGainControl(BGainControl.UP);
+			return;
+		} else if (splitProperty[1].equals(Command.BGAIN.getName() + BGainControl.DOWN.getName())) {
+			bGainControl(BGainControl.DOWN);
+			return;
+		}
+
+		Command imageProcessCommand = Command.getByName(splitProperty[1]);
+		switch (imageProcessCommand) {
+			case WB_MODE:
+				wbModeControl(WBMode.getByName(value));
+				break;
+
+			case WB_ONE_PUSH_TRIGGER: {
+				wbOnePushTrigger();
+				break;
+			}
+			default: {
+				throw new IllegalStateException("Unexpected value: " + Arrays.toString(splitProperty));
+			}
+		}
+
+	}
+
+	/**
+	 * This method is used to control exposure
+	 */
+	private void exposureControl(String value, Command exposureCommand) {
+		switch (exposureCommand) {
+			case BACKLIGHT: {
+				if (value.equals(SWITCH_STATUS_ON)) {
+					backlightControl(BacklightStatus.ON);
+				} else if (value.equals(SWITCH_STATUS_OFF)) {
+					backlightControl(BacklightStatus.OFF);
+				}
+				break;
+			}
+			case AE_MODE: {
+				aeModeControl(AEMode.getByName(value));
+				break;
+			}
+			case EXP_COMP_DIRECT:
+			case GAIN_LIMIT_DIRECT:
+			case GAIN_DIRECT:
+			case IRIS_DIRECT:
+			case SHUTTER_DIRECT: {
+				float directValue = Float.parseFloat(value);
+				directControl(exposureCommand, (int) directValue);
+				break;
+			}
+			case AUTO_SLOW_SHUTTER: {
+				if (value.equals(SWITCH_STATUS_ON)) {
+					slowShutterControl(SlowShutterStatus.ON);
+				} else if (value.equals(SWITCH_STATUS_OFF)) {
+					slowShutterControl(SlowShutterStatus.OFF);
+				}
+				break;
+			}
+			default: {
+				throw new IllegalStateException("Unexpected value: " + exposureCommand);
+			}
+		}
+	}
+
+	/**
+	 * This method is used to control power
+	 */
+	public void powerControl(PowerStatus powerStatus) {
+		try {
+			int currentSeqNum = ++sequenceNumber;
+			byte[] response = send(buildSendPacket(cameraID, currentSeqNum, PayloadType.COMMAND.getCode(), CommandType.COMMAND.getCode(), PayloadCategory.CAMERA.getCode(),
+					Command.POWER.getCode(), powerStatus.getCode()));
+
+			digestResponse(response, currentSeqNum, CommandType.COMMAND, null);
+		} catch (Exception e) {
+			if (this.logger.isDebugEnabled()) {
+				this.logger.debug("error during power " + powerStatus.getName() + " send", e);
+			}
+		}
+	}
+
+	/**
+	 * This method is used to control zoom
+	 */
+	public void zoomControl(ZoomControl zoomControl) {
+		try {
+			int currentSeqNum = ++sequenceNumber;
+			byte[] response = send(buildSendPacket(cameraID, currentSeqNum, PayloadType.COMMAND.getCode(), CommandType.COMMAND.getCode(), PayloadCategory.CAMERA.getCode(),
+					Command.ZOOM.getCode(), (byte) (zoomControl.getCode() + zoomSpeed)));
+
+			digestResponse(response, currentSeqNum, CommandType.COMMAND, null);
+		} catch (Exception e) {
+			if (this.logger.isDebugEnabled()) {
```

Comment:
Is there a reason this error gets swallowed? For control operations it's quite crucial to have some feedback if the operation did not succeed. In this case (if there's an error) it will just silently break without letting user know what went wrong (also on prod debug will be most likely turned off.). 
If it is not acceptable to throw an error here - at least there has to be a warning message. But a rule of thumb - control operation is a request for adapter to do something specific, so it is better to have it explicit

Reply:
Thank you, I will check again